installers/vmware/kickstart-esxi.go

```diff
@@ -231,17 +231,10 @@ fi
 # Kickstart firstboot supplemental config URL
 if [ "$kickstartfburl" != "null" ]; then
 	echo "Using supplemental kickstart firstboot URL: $kickstartfburl"
-	if wget -q -s $kickstartfburl; then
-		wget -q $kickstartfburl -O /tmp/ks-firstboot-sup.sh
-		echo "========Begin execution of supplemental firstboot kickstart"
-		chmod +x /tmp/ks-firstboot-sup.sh && /tmp/ks-firstboot-sup.sh
-		echo "========End execution of supplemental firstboot kickstart"
-	else
-		echo "ERROR: Custom kickstart firstboot URL is NOT accessible!"
-		exit 1
-	fi
-else
-	echo "Skipping supplemental kickstart firstboot URL"
+	wget -q $kickstartfburl -O /tmp/ks-firstboot-sup.sh
```

Comment:
hmm but now you got rid of the handling of `wget` return code, that seems worse right?

Reply:
Possibly :) Should we throw that if check back in then?

Reply:
yeah the if was good, but we didn't need to do the download twice. Could just do the download and branch on success/failure like:
```if ! wget --fail -o ...; then
# handle wget failed.
...
```

Reply:
Unfortunately option "--fail" is either too new or is not supported in the vmware installer. Looking at this further, even though the "-s" option was dropped its still supposed to support "--spider"

Reply:
Unless I'm missing something, `--fail` isn't necessary to do the logic check, wget emits an exit code either way. Here is my personal recommendation:

```shell
wget -q "${kickstartfburl}" -O /tmp/ks-firstboot-sup.sh \
  || { echo  "ERROR: Custom kickstart firstboot URL '${kickstartfburl}' is NOT accessible"; exit 1;}
```

(do note the trailing semi-colon)

alternatively:

```shell
if ! wget -q "${kickstartfburl}" -O /tmp/ks-firstboot-sup.sh; then
  echo  "ERROR: Custom kickstart firstboot URL '${kickstartfburl}' is NOT accessible"
  exit 1
fi
```

I've locally tested both of these options and they appear to function properly. Please let me know if you need any help getting this sorted out.

Reply:
Ah sorry, my familiarity with wget is less than with curl. Does wget return an error if we get a 404 or even 5xx error?

Reply:
Yes, it does in my testing, generally 8:

```
EXIT STATUS
       Wget may return one of several error codes if it encounters problems.

       0   No problems occurred.
       1   Generic error code.
       2   Parse error---for instance, when parsing command-line options, the .wgetrc or .netrc...
       3   File I/O error.
       4   Network failure.
       5   SSL verification failure.
       6   Username/password authentication failure.
       7   Protocol errors.
       8   Server issued an error response.
```

Reply:
Ah nice thanks for looking that up for me. I should have just RTFM :(.